core/arch/arm/mm/mobj_ffa.c

```diff
@@ -15,16 +15,75 @@
 #include <mm/mobj.h>
 #include <sys/queue.h>
 
+/*
+ * Life cycle of struct mobj_ffa
+ *
+ * SPMC at S-EL1 (CFG_CORE_SEL1_SPMC=y)
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ * During FFA_MEM_SHARE allocated in mobj_ffa_sel1_spmc_new() finally added
+ * to the inactive list at the end of add_mem_share() once successfully
+ * filled in.
+ *	registered_by_cookie = false
+ *	mobj.refs.val = 0
+ *	inactive_count = 0
+ *
+ * During FFA_MEM_RECLAIM reclaimed/freed using
+ * mobj_ffa_sel1_spmc_reclaim().  This will always succeed if the normal
+ * world is only calling this when all other threads are done with the
+ * shared memory object. However, there are some conditions that must be
+ * met to make sure that this is the case:
+ *	mobj not in the active list, else -> return TEE_ERROR_BUSY
+ *	mobj not in inactive list, else -> return TEE_ERROR_ITEM_NOT_FOUND
+ *	mobj inactive_count is 0, else -> return TEE_ERROR_BUSY
+ *
+ * mobj is activated using mobj_ffa_get_by_cookie() which unless the mobj
+ * is active already:
+ * - move the mobj into the active list
+ * - if not registered_by_cookie ->
+ *	set registered_by_cookie and increase inactive_count
+ * - set mobj.refc.val to 1
+ * - increase inactive_count
+ *
+ * A previously activated mobj is made ready for reclaim using
+ * mobj_ffa_unregister_by_cookie() which only succeeds if the mobj is in
+ * the inactive list and unregistered_by_cookie isn't set and then:
+ * - sets unregistered_by_cookie
```

Comment:
"the inactive list and **~~un~~**registered_by_cookie is**~~n't~~** set and then:"
"- **~~sets~~ clears** **~~un~~**registered_by_cookie"